src/components/app-root/index.tsx

```diff
@@ -39,6 +42,7 @@ class AppRootBase extends React.PureComponent<AppRootProps, {}> {
     }
     return (
       <div className="voyager">
+        <ClipLoader color={SPINNER_COLOR} loading={dataset.isLoading}/>
```

Comment:
If users already click the load button, it might be better to hide `<LoadData>` and show only the spinner and a text saying "loading".  (There is no reason what we should keep the "load" button.)